avocado/core/safeloader.py

```diff
@@ -250,6 +297,18 @@ def _examine_class(path, class_name):
 
         docstring = ast.get_docstring(klass)
         cl_tags = get_docstring_directives_tags(docstring)
+
+        # Only detect 'avocado.Test' if not yet decided
+        if is_avocado is False:
+            if check_docstring_directive(docstring, 'disable'):
+                is_avocado = True
```

Comment:
Is this correct? In a quick glance it looks like it should be `False` for a class that 
 contains an explicitl `disable`.

Also, every check_docstring_directive() does a lot of the same work.  Maybe it would be better to get all the directives once:

```python
directives = get_docstring_directives(docstring)
if 'disable' in directives:
   is_avocado = ...
```

Reply:
The `is_avocado` means that the class is discovered to be avocado entity. This is very important for recursive discovery (and was buggy in recursive discovery...)

Reply:
... meaning it is avocado entity, it's just not a test.

Reply:
Oh, as for the optimization, sure, could that wait after we get recursive discovery by default?

Reply:
OK, included it in rebased version.